src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,304 @@
+package com.hillel.items_exchange.util;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.hillel.items_exchange.util.PatternHandler.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class PatternHandlerTest {
+
+    @ParameterizedTest
+    @MethodSource("createCorrectEmails")
+    void testEmailRegexp_whenEmailIsCorrect(String correctEmail) {
+        assertTrue(correctEmail.matches(EMAIL));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongEmails")
+    void testEmailRegexp_whenEmailIsWrong(String wrongEmail) {
+        assertFalse(wrongEmail.matches(EMAIL));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectPasswords")
+    void testPasswordRegexp_whenPasswordsCorrect(String correctPassword) {
+        assertTrue(correctPassword.matches(PASSWORD));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongPasswords")
+    void testPasswordRegexp_whenPasswordsWrong(String wrongPassword) {
+        assertFalse(wrongPassword.matches(PASSWORD));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectPhoneNumbers")
+    void testPhoneNumberRegexp_whenPhoneNumberIsCorrect(String correctPhoneNumber) {
+        assertTrue(correctPhoneNumber.matches(PHONE_NUMBER));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongPhoneNumbers")
+    void testPhoneNumberRegexp_whenPhoneNumberIsWrong(String wrongPhoneNumber) {
+        assertFalse(wrongPhoneNumber.matches(PHONE_NUMBER));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectUserNames")
+    void testUsernameRegexp_whenUsernameIsCorrect(String correctUserName) {
+        assertTrue(correctUserName.matches(USERNAME));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongUserNames")
+    void testUsernameRegexp_whenUsernameIsWrong(String wrongName) {
+        assertFalse(wrongName.matches(USERNAME));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsCorrect(String correctWord) {
+        assertTrue(correctWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsWrong(String wrongWord) {
+        assertFalse(wrongWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    private static List<String> createCorrectEmails() {
+        List<String> correctEmails = new ArrayList<>();
+        correctEmails.add("pushkin@ukr.net");
```

Comment:
Why do not you use List.Of() here and elsewhere where you add a lot of elements?

Reply:
I wanted to save possibility add any collections with method .addAll(someList) without rewriting. But you (and drw85, who wrote me about this) are right. I'll rewrite it)).